crates/matrix-sdk/src/event_cache/room/mod.rs

```diff
@@ -925,12 +930,21 @@ mod private {
 
         /// Resets this data structure as if it were brand new.
         #[must_use = "Updates as `VectorDiff` must probably be propagated via `RoomEventCacheUpdate`"]
-        pub async fn reset(&mut self) -> Result<Vec<VectorDiff<TimelineEvent>>, EventCacheError> {
+        pub async fn reset(
+            &mut self,
+            sync_timeline_events_diffs: &mut Vec<VectorDiff<TimelineEvent>>,
```

Comment:
Why do you pass a `&mut` instead of keeping to return the diffs?

Reply:
This was to follow the pattern introduced in `replace_with`, which makes use of this function. I can isolate using the pattern to `replace_with` / `shrink_to_last_chunk`, if you'd prefer, and have the override happen there?

Reply:
I'm struggling to see how it will look like but if the idea is to remove this `&mut`, I think I would prefer that yes!